test/e2e/common/node/container_probe.go

```diff
@@ -609,13 +608,11 @@ done
 
 		// Shutdown pod. Readiness should change to false
 		podClient.Delete(context.Background(), podName, metav1.DeleteOptions{})
-		err = wait.PollImmediate(framework.Poll, f.Timeouts.PodDelete, func() (bool, error) {
-			pod, err := podClient.Get(context.Background(), podName, metav1.GetOptions{})
-			if err != nil {
-				return false, err
+		err = e2epod.WaitForPodStatusByInformer(f.ClientSet, f.Namespace.Name, podName, f.Timeouts.PodDelete, func(pod *v1.Pod) (bool, error) {
+			if !podutil.IsPodReady(pod) {
+				return true, nil
 			}
-			// verify the pod ready status has reported not ready
-			return !podutil.IsPodReady(pod), nil
+			return false, fmt.Errorf("pod %s/%s is still ready", pod.Namespace, pod.Name)
```

Comment:
you don't want to return an error or the loop will exit and fail.
```suggestion
			 framework.Logf("pod %s/%s is still ready", pod.Namespace, pod.Name)
			 return false, nil
```

Reply:
Done